node/src/components/chainspec_loader.rs

```diff
@@ -293,8 +285,26 @@ impl ChainspecLoader {
         self.next_upgrade.clone()
     }
 
+    pub(crate) fn highest_block_header(&self) -> Option<&BlockHeader> {
+        self.highest_block_header.as_ref()
+    }
+
     pub(crate) fn highest_block_hash(&self) -> Option<BlockHash> {
-        self.highest_block_hash
+        self.highest_block_header().map(|hdr| hdr.hash())
+    }
+
+    /// This returns the era at which we will be starting the operation, assuming the highest known
+    /// block is the last one. It will return the era of the highest known block, unless it is a
+    /// switch block, in which case it returns the successor to the era of the highest known block.
+    pub(crate) fn starting_era(&self) -> EraId {
```

Comment:
I agree with Andreas. To me _highest_ is something that can change over time whereas _initial_ is consumed once and will always be the same.

Reply:
> at the point of the most recent upgrade, emergency restart, or genesis.

Except that's not what it means :grimacing: The highest block, as it was called everywhere, is just that: the highest block known when the node was started. This will be genesis at the first start, and the point of the most recent upgrade when the node is started right after an upgrade, but if a node is stopped during normal operation and restarted _without_ an upgrade, this will be just the last block from before the node was stopped, not related to any upgrade at all.

It does define a starting point for some components, so "initial"/"starting" makes sense there, but in essence it's just the highest block when the node is being started.

Having a single term for that would be nice, but I'm not sure if any of these words would fit all the places where it's used.

Reply:
if that's the "highest known block at the moment of startup", and is used to feed the state of the components, I would call it initial.

Reply:
Ah, let's make this `initial` :+1:

Reply:
Right, sorry. :dizzy_face: 
I agree that `initial` would be fine anyway. :+1: